src/java/org/apache/cassandra/transport/Server.java

```diff
@@ -139,18 +139,19 @@ public synchronized void start()
         if (workerGroup != null)
             bootstrap = bootstrap.group(workerGroup);
 
+        final EncryptionOptions clientEnc = DatabaseDescriptor.getNativeProtocolEncryptionOptions();
+
         if (this.useSSL)
         {
-            final EncryptionOptions clientEnc = DatabaseDescriptor.getNativeProtocolEncryptionOptions();
-
-            if (clientEnc.optional)
+            if (clientEnc.tlsEncryptionPolicy() == EncryptionOptions.TlsEncryptionPolicy.optional)
             {
-                logger.info("Enabling optionally encrypted CQL connections between client and server");
+                logger.debug("Enabling optionally encrypted CQL connections between client and server");
                 bootstrap.childHandler(new OptionalSecureInitializer(this, clientEnc));
             }
             else
             {
-                logger.info("Enabling encrypted CQL connections between client and server");
+                logger.debug("Enabling encrypted CQL connections between client and server");
+                assert clientEnc.tlsEncryptionPolicy() == EncryptionOptions.TlsEncryptionPolicy.encrypted : "Unexpected policy: " + clientEnc.tlsEncryptionPolicy();
```

Comment:
Is `clientEnc.tlsEncryptionPolicy()` guaranteed to be non-null?

Reply:
Yes, I can tag it with an `@Nonnull`, as that seems used elsewhere, though I don't know what current best practice is there for static analysis tools.

Reply:
`@NotNull` is the way to go.